src/Terminal/commands/help.ts

```diff
@@ -1,20 +1,104 @@
-import { Terminal } from "../../Terminal";
+import { Player } from "@player";
+import type { AutocompleteData } from "@nsdefs";
 import { TerminalHelpText, HelpTexts } from "../HelpText";
+import { Terminal } from "../../Terminal";
+import { compile } from "../../NetscriptJSEvaluator";
+import { wrapUserNode } from "../../Netscript/NetscriptHelpers";
+import { enums } from "../../NetscriptFunctions";
+import { GetAllServers } from "../../Server/AllServers";
+import { hasScriptExtension, resolveScriptFilePath, ScriptFilePath } from "../../Paths/ScriptFilePath";
+import { Flags } from "../../NetscriptFunctions/Flags";
+import { hasTextExtension } from "../../Paths/TextFilePath";
 
 export function help(args: (string | number | boolean)[]): void {
   if (args.length !== 0 && args.length !== 1) {
     Terminal.error("Incorrect usage of help command. Usage: help");
     return;
   }
   if (args.length === 0) {
-    TerminalHelpText.forEach((line) => Terminal.print(line));
+    const flushedOut = TerminalHelpText.join("\n");
+    Terminal.print(flushedOut);
   } else {
     const cmd = args[0] + "";
     const txt = HelpTexts[cmd];
+
     if (txt == null) {
-      Terminal.error("No help topics match '" + cmd + "'");
+      // Here is where flow lands if we have a player-implemented command
+
+      // Input sanitization
+      const cmdCopy = String(cmd).replace(/^[/.]+/, "") as ScriptFilePath;
+      const filePath = resolveScriptFilePath(cmdCopy);
+
+      const localServer = Player.getCurrentServer();
+
+      if (filePath == null) {
+        if (hasScriptExtension(cmdCopy)) {
+          Terminal.error(`Could not find script '${cmdCopy}'\nMake sure this file exists in this server.`);
+        } else if (hasTextExtension(cmdCopy)) {
+          Terminal.error(
+            `'${cmdCopy}' needs to be either a *.js, *.ts, *.jsx or *.tsx file to have detailed help information.`,
+          );
+        } else {
+          Terminal.error(`No help entry for '${cmdCopy}'.`);
+        }
+        return;
+      }
+
+      const script = localServer.scripts.get(filePath);
+
+      try {
+        if (script == null) {
+          throw new Error("Script pathname has no valid script object");
+        }
+
+        // Deal with help() here
+
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        compile(script, localServer.scripts).then((compiledModule) => {
+          if (compiledModule.help == null) {
+            Terminal.error("No help text for '" + cmdCopy + "'. Implement it by exporting a help() function.");
+            return;
+          }
+
+          const autocompleteData: AutocompleteData = {
```

Comment:
Surely there's a version of this already for autocomplete? They should be shared, unless there's a painful reason why they can't.